modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/categorization/tags/EditTagsModal.es.js

```diff
@@ -0,0 +1,271 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayAlert from '@clayui/alert';
+import ClayButton from '@clayui/button';
+import {ClayRadio, ClayRadioGroup} from '@clayui/form';
+import ClayLoadingIndicator from '@clayui/loading-indicator';
+import ClayModal, {useModal} from '@clayui/modal';
+import EditTagsContext from './EditTagsContext.es';
+import PropTypes from 'prop-types';
+import React, {
+	useCallback,
+	useContext,
+	useEffect,
+	useRef,
+	useState
+} from 'react';
+import {fetch} from 'frontend-js-web';
+import {AssetTagsSelector} from 'asset-taglib';
+
+const EditTagsModal = ({
+	fileEntries,
+	folderId,
+	groupIds,
+	pathModule,
+	repositoryId,
+	selectAll = false,
+	urlSelection = '/bulk/v1.0/bulk-selection',
+	urlTags = '/bulk/v1.0/keywords/common',
+	urlUpdateTags = '/bulk/v1.0/keywords/batch',
+	onModalClose = () => {}
+}) => {
+	const {namespace} = useContext(EditTagsContext);
+
+	const [append, setAppend] = useState(true);
+	const [bulkSelection, setBulkSelection] = useState();
+	const [description, setDescription] = useState('');
+	const [loading, setLoading] = useState(true);
+	const [multiple, setMultiple] = useState(false);
+	const [initialSelectedItems, setInitialSelectedItems] = useState([]);
+	const [selectedItems, setSelectedItems] = useState([]);
+	const [selectedRadioGroupValue, setSelectedRadioGroupValue] = useState(
+		'add'
+	);
+
+	const {observer, onClose} = useModal({
+		onClose: onModalClose
+	});
+
+	useEffect(() => {
```

Comment:
I am not sure we want to use `useEffect` on this one. While the dependencies array gives you an easy way of detecting that something changed between renders, it happens *after* each render. That is, if the props change, we will render with the new props, *then* your `useEffect` callback runs, and because you set state, we re-render *again*. Generally, we should try to avoid double renders when we can (obvious cases when they are necessary are when you have to fetch data, or measure the DOM and then adjust the layout etc). So I think this might be another instance where `usePrevious` could be what you want.